core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2273,7 +2273,7 @@ private[scheduler] class DAGSchedulerEventProcessLoop(dagScheduler: DAGScheduler
 
     case ExecutorLost(execId, reason) =>
       val workerLost = reason match {
-        case SlaveLost(_, true) => true
+        case AgentLost(_, true) => true
```

Comment:
why do we need to introduce a new name "agent" here?

Reply:
Using worker here would be confusing since there is a worker lost flag here. I’m open to other names though for sure :)

Reply:
What does agent mean here?

Reply:
I believe it refers to the host/container level (which can have multiple workers running inside it, although that's deprecated). The documentation of these older message classes is a little sparse though so if you've got a better name I'm all ears.

Reply:
I'm trying to understand what this is before suggesting a name. Is it a process? if yes we can say process ...

Reply:
Yes, it's a process (although many things are processes). I'm happy to use `ProcessLost` if it's clearer than `AgentLost` :)

Reply:
I'd call it ExecutorProcessLost, if that accurately captures what it means.

I wouldn't introduce a new term agent because it can mean a lot of different things.

Reply:
Yeah I think that's a good name for it. I'll refactor that tonight along with any other name changes suggested :)

Reply:
Note: in mesos I'm keeping agent as Mesos it's self is migrating to agent to refer to what used be called Mesos slaves.